spec/operators/pairwise-spec.ts

```diff
@@ -103,4 +106,28 @@ describe('pairwise operator', () => {
     expectObservable(source).toBe(expected);
     expectSubscriptions(e1.subscriptions).toBe(e1subs);
   });
+
+  it('should be recursion-prone', () => {
+    const spy = sinon.spy();
+
+    let emit: ((value: string) => void) | undefined;
+
+    Observable.create((observer: Observer<string>) => {
+      observer.next('a');
+      emit = value => observer.next(value);
+    })
+      .pipe(
+        pairwise(),
+        filter(pair => pair[0] !== pair[1]),
+        take(3)
+      )
+      .subscribe(() => {
+        emit('b');
+        spy();
+      });
+
+    emit('b');
+
+    expect(spy).to.have.been.calledOnce;
```

Comment:
I agree that there is a bug here, but I'm not sure why the spy is expected to have been called only once. And I'd like to see some expectations regarding the values that are emitted.

Basically, I'd like the test to be easier to understand. It could be made a little simpler using a `Subject`. Here's a snippet that reproduces the bug using a `Subject`:

```ts
import { Subject } from "rxjs";
import { filter, pairwise, take } from "rxjs/operators";

const subject = new Subject<string>();
subject.pipe(
  pairwise(),
  filter(pair => pair[0] !== pair[1]),
  take(3)
)
.subscribe(value => {
  console.log(value);
  subject.next("c");
});
subject.next("a");
subject.next("b");
console.log("done");
```

Which outputs:

```
[ 'a', 'b' ]
[ 'a', 'c' ]
[ 'a', 'c' ]
done
```

Reply:
Please have a look now.